### PR TITLE
overlays: Add fbtft overlay

### PR DESCRIPTION
**Pull request by notro**
Add an overlay that provides much of the functionality that fbtft_device did.

Follow up from #4659

### CODE REVIEW

File: arch/arm/boot/dts/overlays/README [hunk 1]
```diff
@@ -857,6 +857,127 @@ Params: interrupt               GPIO used for interrupt (default 4)
         swapxy                  Touchscreen swapped x y axis
 
 
+Name:   fbtft
+Info:   Overlay for SPI-connected displays using the fbtft drivers.
+
+        This overlay seeks to replace the functionality provided by fbtft_device
+        which is now gone from the kernel.
+
+        Most displays from fbtft_device has been ported over.
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: "displays ... _have_ been"

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -857,6 +857,127 @@ Params: interrupt               GPIO used for interrupt (default 4)
...
+        Example:
+          dtoverlay=fbtft,rpi-display,reset_pin=23,dc_pin=24,led_pin=18,rotate=270
+
+        It is also possible to specify the controller (this will use the default
+        init sequence in the driver).
+        Example:
+          dtoverlay=fbtft,ili9341,bgr,reset_pin=23,dc_pin=24,led_pin=18,rotate=270
+
+        The following features of fbtft_device has not been ported over:
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: "features ... _have_ not been"

File: arch/arm/boot/dts/overlays/README [hunk 3]
```diff
@@ -857,6 +857,127 @@ Params: interrupt               GPIO used for interrupt (default 4)
...
+        - it's not possible to change the SPI bus number (spi0 is used)
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: I'd like to make this statement false.

File: arch/arm/boot/dts/overlays/README [hunk 4]
```diff
@@ -857,6 +857,127 @@ Params: interrupt               GPIO used for interrupt (default 4)
...
+        st7789v                 ST7789V display controller
+        tls8204                 TLS8204 display controller
+        uc1611                  UC1611 display controller
+        uc1701                  UC1701 display controller
+        upd161704               UPD161704 display controller
+
+        speed                   SPI bus speed in Hz (default 32000000)
+        cpha                    Shifted clock phase (CPHA) mode
+        cpol                    Inverse clock polarity (CPOL) mode
+        cs                      Chip Select {0, 1} (default 0)
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: I'd like to see this replaced by `spi<m>-<n>` parameters (although it's good to see a lookup table being used in the implementation).

**Review by pelwell — COMMENTED** [review]
Overall this looks great. Would you consider adding support for the other SPI interfaces as well? The `spi<m>-<n>` parameters (where one must be chosen) are now quite common - see the `mcp2515` overlay for an example.

I only mention the tiny grammatical issue in the README because the rest is flawless.

**Comment by notro** [comment]
I've pushed a new version with the requested changes.

I've tried to avoid all the dormant spidev fragements that mcp2515 has, seems to be working.

**Comment by pelwell** [comment]
Thanks!

**Pull request merged**